backend/internal/user/repo/user.go

```diff
@@ -65,7 +66,32 @@ func (r *UserRepo) GetByName(ctx context.Context, username string) (*db.User, er
 }
 
 func (r *UserRepo) ValidateInviteCode(ctx context.Context, code string) (*db.InviteCode, error) {
-	return r.db.InviteCode.Query().Where(invitecode.Code(code)).Only(ctx)
+	var res *db.InviteCode
+	err := entx.WithTx(ctx, r.db, func(tx *db.Tx) error {
```

Comment:
在ValidateInviteCode函数中，使用了数据库事务来确保邀请码的状态更新和检查是原子操作，这提高了代码的可靠性。但需要注意的是，在高并发场景下，这种实现可能导致性能问题。建议考虑使用数据库层面的原子操作或者分布式锁来进一步优化。

```suggestion
// 可以考虑使用数据库的原子操作来替代事务，例如：
// UPDATE invite_codes SET status = 'used' WHERE code = $1 AND status = 'pending' AND expired_at > NOW() RETURNING *